### PR TITLE
Correct the article in generated documents

### LINKED ISSUE #32305: Use "an" instead of "a" before an ingress in auto generated API documentation
In swagger spec, we say "create a ingress", "delete a ingress", etc (https://github.com/kubernetes/kubernetes/blob/943d7aa5aa42d8fc359614e1c02521f48d03b989/api/swagger-spec/extensions_v1beta1.json#L3164)
It should be _an_ ingress.

Relevant code is at https://github.com/kubernetes/kubernetes/blob/943d7aa5aa42d8fc359614e1c02521f48d03b989/pkg/apiserver/api_installer.go#L614

### PR DESCRIPTION
**Pull request by asalkeld**
**What this PR does / why we need it**:
Fix the article in generated docs for "create/delete [article] [kind]"

**Which issue this PR fixes**
fixes #32305

**Special notes for your reviewer**:
None

**Release note**:

``` release-note
Correct the article in generated documents
```

For example:
  "a Ingress" > "an Ingress"

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32557)

### CODE REVIEW

**Comment by brendandburns** [comment]
LGTM, thanks for doing this!

**Comment by pigmej** [comment]
@brendandburns @asalkeld aren't that swagger spec autogenerated? It will be overwritten when anyone will regenerate it, isn't it?

Please see https://github.com/kubernetes/kubernetes/pull/32278#issuecomment-245694206

File: pkg/apiserver/api_installer.go [hunk 1]
```diff
@@ -620,9 +622,10 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 			addParams(route, action.Params)
 			ws.Route(route)
 		case "DELETE": // Delete a resource.
-			doc := "delete a " + kind
+			article := utilstrings.GetArticleForNoun(kind, " ")
+			doc := "delete" + article + kind
 			if hasSubresource {
-				doc = "delete " + subresource + " of a " + kind
+				doc = "delete " + subresource + " of" + article + kind
```
Comment (asalkeld) [inline pkg/apiserver/api_installer.go]: @pigmej here is the actual code change. The other stuff is checked in, but not edited by hand (./hack/update-all.sh)

**Comment by asalkeld** [comment]
Sorry about the noise on this, but I had to rebase and make a bazel BUILD file change.

**Comment by nikhiljindal** [comment]
Thanks a lot @asalkeld for fixing this!
Adding back LGTM.

GCE failure seems to be a flake. Retrying
@k8s-bot gci gce e2e test this

**Comment by brendandburns** [comment]
LGTM (again)

This has got to be the longest small change PR in history... Sorry for the delay!

**Pull request merged**